.github/workflows/reusable-ci.yml

```
@@ -0,0 +1,161 @@
+name: Reusable CI Workflow
+
+on:
+  workflow_call:
+    outputs:
+      remote-mcp-changed:
+        description: "Whether remote-mcp has changes"
+        value: ${{ jobs.change-detection.outputs.remote-mcp }}
+      syllabus-api-changed:
+        description: "Whether syllabus-api has changes"
+        value: ${{ jobs.change-detection.outputs.syllabus-api }}
+      supabase-changed:
+        description: "Whether supabase has changes"
+        value: ${{ jobs.change-detection.outputs.supabase }}
+      workflow-changed:
+        description: "Whether reusable workflow has changes"
+        value: ${{ jobs.change-detection.outputs.workflow }}
+  workflow_dispatch:
+
+jobs:
+  global-ci:
+    runs-on: ubuntu-latest
+    timeout-minutes: 5
+    name: global-ci
+    steps:
+      - uses: actions/checkout@v4
+      - name: Cache turbo build setup
+        uses: actions/cache@v4
+        with:
+          path: .turbo
+          key: ${{ runner.os }}-turbo-${{ github.sha }}
+          restore-keys: |
+            ${{ runner.os }}-turbo-
+      - uses: oven-sh/setup-bun@v2
+        with:
+          bun-version: latest
+      - run: bun install --frozen-lockfile
+      - run: bunx biome lint
+      - run: bunx biome format
+      - run: bun editorconfig-checker
+      - run: bun turbo check:knip
+      - run: bun turbo build
+      - run: bun turbo check:type
+
+  change-detection:
+    runs-on: ubuntu-latest
+    timeout-minutes: 5
+    name: change-detection
+    outputs:
+      remote-mcp: ${{ steps.change-detection.outputs.remote-mcp }}
+      syllabus-api: ${{ steps.change-detection.outputs.syllabus-api }}
+      supabase: ${{ steps.change-detection.outputs.supabase }}
+      workflow: ${{ steps.change-detection.outputs.workflow }}
+    steps:
+      - uses: actions/checkout@v4
+      - uses: dorny/paths-filter@v3
+        id: change-detection
+        with:
+          filters: |
+            remote-mcp:
+              - 'apps/remote-mcp/**'
+            syllabus-api:
+              - 'apps/syllabus-api/**'
+            supabase:
+              - 'apps/syllabus-api/supabase/**'
+            workflow:
+              - '.github/workflows/reusable-ci.yml'
```

```suggestion
          filters: |
            remote-mcp:
              - 'apps/remote-mcp/**'
              - '.github/workflows/reusable-ci.yml'
            syllabus-api:
              - 'apps/syllabus-api/**'
              - '.github/workflows/reusable-ci.yml'
            supabase:
              - 'apps/syllabus-api/supabase/**'
              - '.github/workflows/reusable-ci.yml'
```
こうすると、`.github/workflows/reusable-ci.yml`が変更された時に、PRでもciが回って動作確認できるよね？